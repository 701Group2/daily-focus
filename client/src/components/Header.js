import React from "react";
import Button from "@material-ui/core/Button";
import { makeStyles } from "@material-ui/core";

import styles from "./Header.module.scss";
import FocusLogo from "../images/focus-logo.svg";

const useStyles = makeStyles({
    root: {
        background: "#30A0F5",
        borderRadius: 50,
        padding: "4px 25px",
    },
    label: {
        textTransform: "capitalize",
        fontSize: "1.25rem",
    },
});

export default function Header() {
    const classes = useStyles();

    return (
        <header className={styles.header}>
            <div className={styles.focusLogo}>
                <img style={{ width: 175 }} src={FocusLogo} alt="FOCUS" />
            </div>

            <div className={styles.dateTime}>10:09 AM | Monday 8 March</div>

            <div className={styles.searchBar}>Google Search</div>

            <div className={styles.addWidget}>
                <Button
                    variant="contained"
                    color="primary"
                    classes={{ root: classes.root, label: classes.label }}
                >
                    Add Widget
                </Button>
            </div>

            <div className={styles.userIcon}>:D</div>
        </header>
    );
}
