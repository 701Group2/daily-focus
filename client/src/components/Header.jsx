import React from "react";

import styles from "./header.module.scss";

export default function Header() {
    return <header className={styles.header}>
        <div className={styles.focusLogo}>
            FOCUS
        </div>

        <div className={styles.dateTime}>
            10:09 AM | Monday 8 March
        </div>

        <div className={styles.searchBar}>
            Google Search
        </div>

        <div className={styles.addWidget}>
            Add Widget
        </div>

        <div className={styles.userIcon}>
            :D
        </div>
    </header>;
}
