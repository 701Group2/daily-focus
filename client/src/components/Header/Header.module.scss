.header {
    display: flex;
    height: 100px;
    justify-content: space-around;
    align-items: center;
    font-size: 1.5rem;
    background-color: white;
    -webkit-box-shadow: 0px 7px 20px 0px rgba(0, 0, 0, 0.1);
    -moz-box-shadow: 0px 7px 20px 0px rgba(0, 0, 0, 0.1);
    box-shadow: 0px 7px 20px 0px rgba(0, 0, 0, 0.1);

    .focusLogo {
        flex: 1 1 0;
        text-align: center;
    }

    .dateTime {
        flex: 1 1 0;
        color: #7d7d7d;
    }
    .searchBar {
        flex: 1.5 1 0;
        border: 2px solid #7d7d7d;
        color: #7d7d7d;
        border-radius: 20px;
        padding: 4px 0px 4px 20px;
    }

    .addWidget {
        flex: 0.75 1 0;
        text-align: center;
    }

    .userIcon {
        flex: 0.4 2 0;
    }
}
