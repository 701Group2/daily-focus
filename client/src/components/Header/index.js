import React, { useState } from "react";
import { IconButton, makeStyles } from "@material-ui/core";
import PersonIcon from "@material-ui/icons/Person";
import styles from "./style.module.scss";
import FocusLogo from "../../images/focus-logo.svg";
import SearchBar from "../SearchBar";
import DateTime from "../DateTime";
import SelectWidgetsModal from "../SelectWidgetsModal";

const useStyles = makeStyles({
    root: {
        background: "#30A0F5",
        borderRadius: 50,
        padding: "4px 25px",
    },
    label: {
        textTransform: "capitalize",
        fontSize: "1.25rem",
    },
    icon: {
        "background": "#30A0F5",
        "padding": 10,
        "&:hover": {
            backgroundColor: "#303f9f",
        },
    },
    iconLabel: {
        fill: "white",
        fontSize: "2rem",
    },
});

export default function Header({ selectedWidgets, setSelectedWidgets }) {
    const classes = useStyles();
    const [expand, setExpand] = useState(false);

    return (
        <header className={styles.header}>
            <div className={styles.focusLogo}>
                <img style={{ width: 175 }} src={FocusLogo} alt="FOCUS" />
            </div>
            <div
                className={expand ? styles.collapseIcon : styles.collapseIconExpanded}
                onClick={() => setExpand(!expand)}
            >
                <div className={styles.collapseBar}></div>
                <div className={styles.collapseBar}></div>
                <div className={styles.collapseBar}></div>
            </div>
            <div className={expand ? styles.headerContainer : styles.headerContainerExpanded}>
                <div className={styles.dateTime}>
                    <DateTime />
                </div>

                <div className={styles.searchBar}>
                    <SearchBar />
                </div>

                <div className={styles.addWidget}>
                    <SelectWidgetsModal
                        selectedWidgets={selectedWidgets}
                        setSelectedWidgets={setSelectedWidgets}
                    />
                </div>

                <div className={styles.userIcon}>
                    <IconButton aria-label="user menu" classes={{ root: classes.icon }}>
                        <PersonIcon classes={{ root: classes.iconLabel }} />
                    </IconButton>
                </div>
            </div>
        </header>
    );
}
