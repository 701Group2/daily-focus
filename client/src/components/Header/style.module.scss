.header {
    position: sticky;
    top: 0;
    z-index: 10;
    display: flex;
    justify-content: space-between;
    align-items: center;
    font-size: 1.5rem;
    background-color: white;
    padding: 10px;
    -webkit-box-shadow: 0px 7px 20px 0px rgba(0, 0, 0, 0.1);
    -moz-box-shadow: 0px 7px 20px 0px rgba(0, 0, 0, 0.1);
    box-shadow: 0px 7px 20px 0px rgba(0, 0, 0, 0.1);

    .collapseIcon {
        visibility: hidden;
        position: absolute;
        z-index: 99;
        top: 5px;
        right: 10px;
        transform: translateX(-50%);
        width: 40px;
        height: 50px;
        display: flex;
        flex-direction: column;
        justify-content: center;
        cursor: pointer;

        &Expanded {
            @extend .collapseIcon;

            & > .collapseBar:nth-child(1) {
                transform: rotate(-45deg);
            }
            & > .collapseBar:nth-child(2) {
                opacity: 0;
                transform: translateX(100%);
            }
            & > .collapseBar:nth-child(3) {
                transform: rotate(45deg);
            }
        }

        .collapseBar {
            width: 100%;
            height: 4px;
            margin: 5px 0;
            background-color: #30a0f5;
            transition: all 0.25s ease;
            transform-origin: right;
        }
    }

    .headerContainer {
        display: flex;
        justify-content: flex-end;
        align-items: center;

        &Expanded {
            @extend .headerContainer;
        }

        & > div {
            margin: 5px;
        }

        .focusLogo {
            flex: 0 1 auto;
            text-align: center;
        }

        .dateTime {
            flex: 0 1 auto;
        }
        .searchBar {
            flex: 0 1 auto;
        }

        .addWidget {
            flex: 0 1 auto;
            text-align: center;
        }

        .userIcon {
            flex: 0 1 auto;
        }
    }

    @media (max-width: 800px) {
        .collapseIcon {
            visibility: visible;
        }
        .headerContainer {
            padding-right: 15px;
            padding-left: 5px;
            visibility: hidden;
            position: absolute;
            top: 0;
            right: 0;
            height: 100vh;
            flex-direction: column-reverse;
            background-color: white;
            border-left: solid 10px #0f7ccf;
            box-sizing: border-box;
            justify-content: flex-start;
            align-items: center;
            transform: translateX(100%);
            transition: transform 0.5s ease, visibility 0s ease 0.5s;

            &Expanded {
                visibility: visible;
                transform: translateX(0);
                transition: transform 0.5s ease;
            }

            .dateTime {
                margin-bottom: 20px;
            }
        }
    }
}
