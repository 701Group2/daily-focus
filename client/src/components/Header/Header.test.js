import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import Header from ".";

test("it renders correctly", () => {
    const renderer = new ShallowRenderer();
    const component = renderer.render(<Header />);
    expect(component).toMatchSnapshot();
});
