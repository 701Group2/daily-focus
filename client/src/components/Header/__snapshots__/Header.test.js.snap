// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="focusLogo"
  >
    <img
      alt="FOCUS"
      src="focus-logo.svg"
      style={
        Object {
          "width": 175,
        }
      }
    />
  </div>
  <div
    className="dateTime"
  >
    <DateTime />
  </div>
  <div
    className="searchBar"
  >
    <SearchBar />
  </div>
  <div
    className="addWidget"
  >
    <SelectWidgetsModal />
  </div>
  <div
    className="userIcon"
  >
    <WithStyles(ForwardRef(IconButton))
      aria-label="user menu"
      classes={
        Object {
          "root": "makeStyles-icon-3",
        }
      }
    >
      <Memo
        classes={
          Object {
            "root": "makeStyles-iconLabel-4",
          }
        }
      />
    </WithStyles(ForwardRef(IconButton))>
  </div>
</header>
`;
